### PR TITLE
App prefernces caching.

### PR DESCRIPTION
**Pull request by TheSenPie**
Apps can now persist their preferences per user, project, app and game asset in a simple manner.

### CODE REVIEW

File: editor/src/com/talosvfx/talos/editor/project2/apps/SceneEditorApp.java [hunk 1]
```diff
@@ -86,5 +86,27 @@ public void onUpdate () {
...
+	@Override
+	public Object getCurrentPreference() {
+		SceneEditorAppPrefs prefs = (SceneEditorAppPrefs) super.getCurrentPreference();
+		prefs.cameraPos = workspaceWidget.getCameraPos();
+		prefs.cameraZoom = workspaceWidget.getCameraZoom();
+		return prefs;
+	}
+
+	@AppPreference
+	private static class SceneEditorAppPrefs {
```
Comment (Tom-Ski) [inline editor/src/com/talosvfx/talos/editor/project2/apps/SceneEditorApp.java:107]: This should extend some other class. ViewportViewSettings are in many apps, and it doesn't make sense to add this stuff just for SceneEditorApp and then copy paste to others

File: editor/src/com/talosvfx/talos/editor/project2/apps/AppPrefs.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        return preferences.get(getUUID(gameAsset));
+    }
+
+    public void setPrefFor(GameAsset<?> gameAsset, Object appPref) {
+        preferences.put(getUUID(gameAsset), appPref);
+    }
+
+    private String getUUID (GameAsset<?> gameAsset) {
+        if (!gameAsset.dependentRawAssets.isEmpty()) {
+            return gameAsset.getRootRawAsset().metaData.uuid.toString();
```
Comment (Tom-Ski) [inline editor/src/com/talosvfx/talos/editor/project2/apps/AppPrefs.java:30]: Also, with the fake game asset, we can probably make a fake dependent raw asset too, s we don't need this check

Comment (Tom-Ski) [reply editor/src/com/talosvfx/talos/editor/project2/apps/AppPrefs.java:30]: Ive just done this for one of my tasks, so you can use the same

File: editor/src/com/talosvfx/talos/editor/project2/apps/SceneEditorApp.java (hunk 1 above)
Comment (TheSenPie) [reply editor/src/com/talosvfx/talos/editor/project2/apps/SceneEditorApp.java]: Done

File: editor/src/com/talosvfx/talos/editor/project2/apps/AppPrefs.java (hunk 2 above)
Comment (TheSenPie) [reply editor/src/com/talosvfx/talos/editor/project2/apps/AppPrefs.java]: Done

**Review by Tom-Ski — APPROVED** [review]

**Pull request merged**